crowbar_framework/app/models/crowbar_service.rb

```diff
@@ -385,9 +385,14 @@ def prepare_nodes_for_os_upgrade
     upgrade_nodes.each do |node|
       node["target_platform"] = admin_node["provisioner"]["default_os"]
       node["crowbar_wall"]["crowbar_upgrade_step"] = "prepare-os-upgrade"
+      # skip initial keystone bootstrapping
+      node["keystone"]["bootstrap"] = true if node["keystone"]
       node.save
     end
 
+    # unset `db_synced` flag for OpenStack components
+    ::Openstack::Upgrade.unset_db_synced
```

Comment:
I don't think there's an option to cancel the upgrade at this point (which is called "stopping the services" in the UI). This is already Cloud7 side.